### ISSUE TITLE
Mouse Y coordinate sometimes skips by 2 during movement

### ISSUE THREAD

**Issue by dansanderson** [MEGA65/mega65-rom-public#90]
**Test Environment (required)**
You can use [MEGA65INFO](https://files.mega65.org?id=3eeb7d82-8bd7-44b7-82b6-cef29d123b0e) to retrieve this.
 - Platform: MEGA65
 - ROM Release: 920389 and not 920377 (regression since previous stable)
 - Core Commit: Tested with v0.95 stable core and latest dev core

**Describe the bug**
RMOUSE's value for the button variable has bit 1 set incorrectly, affected by changes to the Y coordinate. This only occurs with the mouse in port 1.

Specifically, if I move the mouse slowly downward, the Y coordinate appears to be missing bit 0, and bit 0 appears on the button variable: row 0 is MY=50 MB=0, row 1 is MY=50 MB=1, row 2 is MY=52 MB=0, and so on. MY sometimes gets odd-value increments when moving the mouse more quickly or in the middle of the screen.

Note that I *can* reproduce the Y skipping behavior in ROM 920377 and in both ports. The change in a later ROM adds a bit erroneously to the button variable, and only in port 1. The stray bit looks like it would correct the skipping behavior, so it might have been an incomplete attempt at a fix.

I cannot reproduce this in 920377, so this appears to be a recent regression, probably during the screen system refactor for 80x50 support. I can reproduce this in both the v0.95 stable core and the latest dev core, so it does not appear to be related to mouse-related core changes. The latest dev core repairs mouse support in port 2, and the latest ROM does not have the stray bit behavior for port 2.

Affects both 1351 and Amiga mice.

**To Reproduce**
```
10 MOUSE ON,1
20 RMOUSE MX,MY,MB
30 PRINT MX,MY,MB
40 GOTO 20
```

**Additional context**
Mouse in port 1 active also interferes with the screen editor in other ways. It's easy to get the cursor to disappear. For example, this test program will fail in a bad way when Run/Stop is pressed, but only if it's `MOUSE ON,1`. With `MOUSE ON,2`, it does not have this issue. This bug should be filed separately but I mention it here in case it's related.

**Comment by johnwayner** [comment]
With a core from 11/23/23 and the latest Beta ROM, I'm only able to reproduce the button bit 0 on y-movement issue on either port if my MouSTer or wireless Tank Mouse* is configured for Amiga mouse mode.  I don't have an actual Amiga mouse, but I would expect the same behavior from it, too.

A key difference between the wiring of an Amiga mouse vs a 1351 is that the vertical pulse is connected to the same pin that a 1351's right mouse button is connected to.  So it's not surprising that the right mouse button would appear to change when moving vertically.  But I would only expect an actual Amiga mouse or a mouse configured to act like one to have this issue.

In the core, I'm only finding code to modify how the mouse MOVEMENT is modified if Amiga mouse is configured or detected. I see nothing to override the driving of the CIA port values to different values if an amiga mouse is detected.  I'm out of VHDL depth here, however.  Maybe it's not possible to modify the CIA port value reported for the buttons without breaking joysticks?

So, @dansanderson , I know my results don't line up with what you've put in this issue.  Given the above information about how the Amiga mouse is wired up, can you retest with that in mind?  I wonder if a red herring colored your interpretation of your previous results.

*Note about the wireless Tank Mouse dongle in Amiga mode.  It doesn't seem to work with the Amiga to 1351 configuration. I can only get the Tank to work in 1351 mode.  And even then the DPI is set way too high and is not configurable :( The Tank works fine with the MouSTer (instead of the provided dongle) with a divisor of about 15 in 1351 mode.

As for the Y skipping behavior, I have a fix for that.  It's an entirely separate issue in the kernal mouse driver.

I'm also not seeing any issues when running on port 1 and hitting Run/Stop.  Maybe it's been fixed already or it's an intermittent issue.

**Comment by dansanderson** [comment]
I concur that I'm not able to repro a few of the details in my original report, and that I may be reporting several unrelated issues. (I piled them all in because I wasn't sure if they were related at the time.) I can only repro with Amiga-style mice, not the 1351, which matches your experience.

Your Y-skipping fix will be welcome. I was able to repro your fix with your build.

That's really interesting info about the Amiga mouse pulsing the 1351 right mouse button signal! That suggests that the Amiga mouse mode of the firmware needs to account for this. As I understand it, without a core fix, there is no other way to read the right mouse button, so this is genuine interference. I'll file a core bug for that.

I'm definitely still seeing some erratic behavior in the screen editor (disappearing cursor being one symptom) with a mouse connected to port 1. This is most prominent when the mouse configuration is in the wrong mode, and I normally treat that as expected behavior. But I also seem to see it occasionally with an Amiga mouse in port 1 configured in Amiga mode. I'm not able to repro it reliably, and I wonder if it's a mild ROM interaction like something flipping the "cursor on" flag in the KERNAL, vs. an irrevocable hardware shortcoming like the classic joystick port 1-keyboard CIA interference. We may just have to keep our eyes peeled for issues.

Let's make this bug specifically about the Y skipping, and resolve it once you have a final fix. Thanks very much!
